### ISSUE TITLE
Blazor project tracking 2021

### ISSUE THREAD

**Issue by guardrex** [dotnet/AspNetCore.Docs#19286]
This issue will ...

* Maintain a set of doc ideas for Blazor topics that we either don't want to (or can't) address in the near-term or are just ideas for consideration that don't warrant an issue at this time.
* Track work on topic UE (user experience) passes (i.e., topic overhauls).

### Doc ideas

* [ ] Confirm with the PU that runtime relinking is a **publish-and-Release config** gesture. Building only produced 51.5 MB. Publishing in `Debug` was 9.3 MB without relevant console compiler output. Publishing in `Release` was 6.6 with console compiler output that looked like runtime relinking was taking place.
* [ ] When https://github.com/dotnet/docs/issues/27395 is resolved and Mono/WASM MSBuild props are covered, update the cross-links from the targets file to the new official doc. Links can be located on the PR that added them at https://github.com/dotnet/AspNetCore.Docs/pull/24493.
* [ ] It seems like everyplace that we have hosted WASM, we should call out that the project must be run from the **`Server`** project. The errors for not doing so can be quite cryptic 😵. For example, the File Uploads topic example throws a "JSON token" error when the solution is run from the client app. It's not at all clear that it was simply run from the wrong project.
* [x] The pivots for the *Handle Errors* topic were dropped, then the content was reorganized on https://github.com/dotnet/AspNetCore.Docs/pull/24195 for its final layout. However, the *Prerendering* section is currently only for Blazor Server apps. That section should be made to generically cover both WASM and Server prerendering. https://github.com/dotnet/AspNetCore.Docs/pull/24211
* [x] For `NavLink` components in examples: It will harden docs against past and future deltas to merely mention adding example components to the `NavMenu` and not actually show the markup, which seems to change quite a bit. #24130
* [x] A number of spots show project file XML markup for adding packages to apps. There are also Package Manager and CLI commands. I think mentioning the package would be better and having an INCLUDE for the options: edit the project file, use the Package Manager, or use the CLI add package command. I should be able to filter the locations for these updates with `<PackageReference`. https://github.com/dotnet/AspNetCore.Docs/pull/24212
* [ ] In the *Prerender and Integrate* topic, clarify the line that speaks to the dev starting with RP/MVC and then including Blazor Server app in order to render components from pages/views. https://github.com/dotnet/AspNetCore.Docs/issues/23948#issuecomment-974559793 **Related**: I add a `HeadOutlet` component to the layout at Line 212 for the scenario of an RP/MVC app that embeds components from a hosted WASM app (`render-mode="ServerPrerendered"`) to allow components to control head content. Due to time constraints at the moment, the coverage is going in. When I address this item and touch that line :point_up: also run a quick test and confirm the scenario for Line 212 actually works ... *or did I just make up a new Blazor feature all by myself?!* 🙈😄
* [X] Drop the images added by https://github.com/dotnet/AspNetCore.Docs/pull/23959. They're provided by the PU for quick coverage, but we don't want images of the Azure portal unless absolutely necessary given the UI churn of the portal. Also, improve the naming (e.g., app name) ... and this will require text changes (e.g., `https://bwhappsettingsasdf-staging.azurewebsites.net` is in the doc text). https://github.com/dotnet/AspNetCore.Docs/pull/24234
* [ ] `InputFile` component preview guidance (6.0) is in the new *Blazor Images* topic and probably should be better cross-linked. https://github.com/dotnet/AspNetCore.Docs/issues/23802
* [x] Drop hosting model pivots from a few topics where they aren't needed, especially in *Fundamentals* node topics b/c readers are probably consuming all of these docs early and without regard to an explicit choice of hosting model. https://github.com/dotnet/AspNetCore.Docs/pull/23736 [PR for Call web API Demo 3 test (live now)](https://github.com/dotnet/AspNetCore.Docs/pull/23753) & [Additional updates for Call web API](https://github.com/dotnet/AspNetCore.Docs/pull/23755)
* [x] Sub-app hosting: Let's firm up a prohibition on the use of IIS virtual folders and firm up recommendation to host separate apps in separate app pools, including for sub-apps. https://github.com/dotnet/AspNetCore.Docs/pull/24235
* [x] Clarify use case/config for [Blazor Server `MapFallbackToPage` configuration](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/?view=aspnetcore-5.0&tabs=visual-studio#blazor-server-mapfallbacktopage-configuration). Started on: https://github.com/dotnet/AspNetCore.Docs/pull/23698 Finished up on (but it's rather **_draftish_** and needs more work later): #23735 Later, ask about https://github.com/dotnet/aspnetcore/issues/32333#issuecomment-831159066 relative to what I did here with my ordering of the fallbacks. [PU Issue](https://github.com/dotnet/aspnetcore/issues/19744) [PU Issue (vaguely related without full discussion)](https://github.com/dotnet/aspnetcore/issues/21574)
* [ ] Ask Pranav about https://github.com/dotnet/AspNetCore.Docs/pull/23566#issuecomment-947503152.
* [x] Test some solutions for the **_content-hidden-in-a-pivot_** and **_search-results-don't-link-to-visible-content_** problems:
  * [Demo 1](https://github.com/dotnet/AspNetCore.Docs/pull/23554): Show version using the old style, where each section calls out what hosting model is applies to.
  * [Demo 2](https://github.com/dotnet/AspNetCore.Docs/pull/23555): Show a stacked H1 version where the hosting model is the H1 section and it's following sections are H2s or lower.
  * [Demo 3](https://github.com/dotnet/AspNetCore.Docs/pull/23556): Show a version where the section content of each pivot is displayed at the top of the alternate pivot.
* [ ] Ask RA for an internal issue on pivot search results. The problem is that the whole topic is indexed but the links to the topics don't point to the pivot where the content lives ... the topic loads with whatever default pivot the user has selected earlier (or the default). I think the search results should be by-pivot with a clear pivot indication and with a pivot-specific landing link. This would mean that it's possible for the reader to receive two or more search result links, but the reader would be taken directly to the pivot and not get confused when the land to find *no content* for what they searched.
* [ ] For Blazor AOT, add remark ... https://github.com/dotnet/aspnetcore/issues/37092#issuecomment-930356311 ... also see :point_right: https://github.com/dotnet/aspnetcore/issues/35302
* [ ] Ping Steve to take a look at the new guidance on https://github.com/dotnet/AspNetCore.Docs/pull/23223. The guidance+example might be incorrect, but the pattern was signed off by Artak at https://github.com/dotnet/aspnetcore/issues/17886#issuecomment-566158668. I tried not to *totally make something up* 😆, but I may have done so in this case. Ping back Darragh Jones on the outcome.
* [x] Confirm that in spots where we're running a hosted WASM project in a command shell that the **Server** project is where the dev runs the app (https://github.com/dotnet/AspNetCore.Docs/issues/23166). Although VS gets a fix (Preview 4, https://github.com/dotnet/aspnetcore/issues/35356) not to default the startup project to the **Client** project for VS 2022, confirm it with VS 2022 at GA. **Update**: Done! #23227
* [x] Explicit example of lazy loading (e.g., in the snippet sample apps). I'm rejecting on the grounds that its a straight-forward, well-described scenario in the topic and there isn't a lot of time to include it in the snippet sample app. #20154
* [x] Check on the feature request for `Support WebSocket connections on Azure Front Door` in the *Threat Mitigation* topic added on https://github.com/dotnet/AspNetCore.Docs/pull/23033: Update the text and/or kill the link when they either close it out as a `won't-fix` or decide to proceed with implementing the feature. #23279
* [x] Dashes in filenames for Security node projects. See https://github.com/dotnet/aspnetcore/issues/35337#issuecomment-900459109. [Docs PR](https://github.com/dotnet/AspNetCore.Docs/pull/23119)
* [x] Check on encoding in query strings: We want `%20` for spaces everywhere. **Checked 8/25**: No instances found to update.
* [x] Bit concerned about 3.1 (not 3.2 ... 3.1) coverage for Blazor loc. The MSBuild prop isn't involved, and framework churn might have led to poor guidance for 3.1. It will be low priority, but I'll check if I can find time 🏃😅. **Update 8/25**: Checking this one off as a `wont-fix`: Given the arrival of 6.0 soon and the sunset of 3.1 LTS in 12/2022, this is a low enough priority item to merely see if reader feedback indicates a problem with 3.x coverage.
* [x] Target-typed new expressions. See :point_right: https://devblogs.microsoft.com/dotnet/announcing-net-5-0-preview-8/#target-typed-new-expressions &mdash; First pass on https://github.com/dotnet/AspNetCore.Docs/pull/21934. Additional opportunities will be addressed as I go forward with UE passes after 3/31/21.
* [x] Move the *Layouts* topic into the *Components* node? **_UPDATE_**: *Done ...* https://github.com/dotnet/AspNetCore.Docs/pull/22368
* [x] `ShouldRender` referenced in the *Rendering* topic looks to be based in the perf topic, also appears in the *Lifecycle* topic, but perhaps would be better in the *Rendering* topic itself with cross-links from perf and *Lifecycle* topics. https://github.com/dotnet/AspNetCore.Docs/pull/22475
* [x] Call out heuristic scanner false positives with an INCLUDE. We've had ~3 issues now opened on (probably) false positives (for example, https://github.com/dotnet/AspNetCore.Docs/issues/21829). Include strong lingo that it's the dev's responsibility to manage scanner exceptions, since we can't really know if the local copy of a file was modified after it was pulled down from somewhere where the source isn't infected. Also, there's something at https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-5.0#troubleshoot-integrity-powershell-script to address for this item. PR: https://github.com/dotnet/AspNetCore.Docs/pull/22338
* [x] The MVC scenarios in the *Prerender and Integrate* topic weren't fully tested due to time constraints. Perform local testing of those aspects to confirm the guidance. **Update 8/25**: Checking this one off here because it's going to be tracked by the [6.0 tracking issue](https://github.com/dotnet/AspNetCore.Docs/issues/22045). I plan to verify all scenarios in the doc for 6.0 at RC1 and not further test prior releases unless a dev writes in with a problem with 3.1 or 5.0.
* [x] Review https://github.com/dotnet/AspNetCore.Docs/issues/21277 in the context of all Blazor doc set docs. Originally, components could only be used in Blazor apps (and the content was written that way), but they can be used in ASP.NET Core apps now. **UPDATE (7/19)**: We won't be able to address this unless the PU were to move Razor components away from WASM/Server ... e.g., they become part of Razor formally and not just be based on Razor syntax ... which doesn't seem like its going to happen. Components will be Blazor framework objects made available to other frameworks in parallel with those frameworks, so they'll continue to be documented that way AFAICT. The *Components* node will stay in the *Blazor* node. The inclusion of "Blazor" as a web tech tho is a bit funky now tho with MAUI, which doesn't have to be a web tech.
* [x] Re-evaluate the RCL topic for snippet app use. *Overkill, or good i-dear?* **_UPDATE_**: *Done ...* Let's not given the workload. It can be done the next time the RCL topic is overhauled in a few years.
* [x] Document that `<head>` element interaction is only supported via JS interop? See: https://github.com/dotnet/AspNetCore.Docs/issues/22121. **_UPDATE_**: *Done ...* This has been placed on the 6.0 tracking issue to resolve when the `<head>` element features are added to the doc set (see: https://github.com/dotnet/AspNetCore.Docs/issues/22045).
* [x] Take a look at https://docs.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-5.0#comparison-to-server-rendered-ui and further sections in light of the discussion on https://github.com/dotnet/AspNetCore.Docs/issues/22165. There are probably opportunities for improving the coverage that compares and contrasts RP/MVC and Blazor frameworks. **Update 8/25**: Checking off because this is going to be tracked by https://github.com/dotnet/AspNetCore.Docs/issues/23120 as part of the 6.0 updates.
* [x] Address the use of Bootstrap JS in a suitable spot. We warn off of interacting with the DOM in the JS interop topic, but Bootstrap guidance (for styles) appears earlier in the doc set. It's bound to be a question in devs' minds. See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/22213. https://github.com/dotnet/AspNetCore.Docs/pull/22489
* [ ] Create an *Overview* in the *Fundamentals* node with a bare bones explanation of a component in order to understand fundamental concepts+examples better, orient on any doc consumption features (e.g., use of working examples, sample apps, use of the `this` keyword, resolve #14216, etc.), and any other prelim remarks before the reader gets into the meat of the coverage. https://github.com/dotnet/AspNetCore.Docs/issues/22230#issuecomment-875910670
* [x] Consistency of Razor component directive order &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/22357
* [x] Confirm/update topic redirection is correct (e.g., check for double-redirection) &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/22358
* [x] WRT the early use of the snippets sample app: I'm not favoring using partial components that require mocked pieces and inlined `<snippet>` tags. This approach is generating a lot of local DocFX build warnings that are clouding real (important) warnings that I need to address. It's slowing me down 🐌. Either move these snippets back into the topics or recompose them or relink them without the use of `<snippet>` tags in their markup. &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/22359
* [x] Confirm null check in object disposal (e.g., `objRef?.Dispose();`). https://github.com/dotnet/AspNetCore.Docs/pull/22597
* JS interop
  * [ ] Call JS from .NET doesn't demo cancellation token use. I think a working cut-'n-paste example wouldn't take up much space and would be nice to show.
  * [ ] The *Call .NET from JS* topic mentions but doesn't show an example of calling an open generic method ... include one? Cross-ref: [PU `InteropComponent.razor`](https://github.com/dotnet/AspNetCore/blob/main/src/Components/test/testassets/BasicTestApp/InteropComponent.razor) and [`GenericType<TValue>`](https://github.com/dotnet/aspnetcore/blob/main/src/Components/test/testassets/BasicTestApp/InteropTest/JavaScriptInterop.cs#L507)
  * [x] The JS interop topics have no guidance on reporting explicit exceptions from JS (e.g., wrap the invoke and then `catch (JSException e)`) ... include a section and one or more examples? Cross-ref: [PU `InteropComponent.razor`](https://github.com/dotnet/AspNetCore/blob/main/src/Components/test/testassets/BasicTestApp/InteropComponent.razor) Docs PR: #22772
  * [ ] Is there anything else in the family of JS interop JS function calls that should be mentioned or explained out with examples? Cross-ref: [PU `Microsoft.JSInterop.ts`](https://github.com/dotnet/aspnetcore/blob/main/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts)
  * [x] React to https://github.com/dotnet/AspNetCore.Docs/pull/19734#issuecomment-683922820. **Update**: A fix was made at the time, and no further negative feedback has been received. AFAIK, the sections+examples are composed correctly.
* [x] "server API" and "web API" are both in use across Blazor docs. Stabilize on "web API" to match general terminology in ASP.NET Core docs. https://github.com/dotnet/AspNetCore.Docs/pull/22410
* [x] For the file upload topic (and perhaps the one in the main doc set, too), consider adding a blurb about how PS can create dummy test files. https://github.com/dotnet/AspNetCore.Docs/pull/22591
* [x] Confirm/add null checks on element references (`ElementReference`, `@ref`) before calling methods on them. https://github.com/dotnet/AspNetCore.Docs/pull/22771
* [ ] Obtaining v2.0 JWT tokens. See convo at :point_right: https://github.com/dotnet/aspnetcore/issues/35794#issuecomment-906809847
* [ ] Old sample apps dropped on #23228. Do we want to make the snippet sample apps into running apps and link them in the Blazor docs? Related ?: Whether or not we make snippet samples for download, should sample apps in general be kept/cross-linked in the official MS samples repo for easy download. This was a long-range plan for all ASP.NET Core sample apps AFAIK, but it hasn't taken place due to constant higher-priority work. We could at least make the Blazor Server-EF Core final app and the two Blazor-SignalR tutorial final apps available via that mechanism for easy download.
* [ ] Do we want to make the *Static files* topic hosting model agnostic and get into static asset caching behavior, including how to control/manage it during development. Alternatively, we could document caching behaviors/scenarios in the *Environments* topic at https://docs.microsoft.com/aspnet/core/blazor/fundamentals/environments. The (possibly temporary) PR that focuses on JS file caching went in on https://github.com/dotnet/AspNetCore.Docs/pull/23235.

### UE pass tracking

🔴 = Snippets update complete

- [x] Overview &mdash; 🔴 
- [x] Supported platforms 🔴
- [x] Tooling &mdash; 🔴
- [x] Hosting models &mdash; #20868 🔴
- [x] Tutorials
  - [x] Build a Blazor todo list app &mdash; #20941 🔴
  - [x] SignalR with Blazor WebAssembly &mdash; #21271 🔴
- [x] Project structure &mdash; #21247 🔴
- [x] Fundamentals
  - [x] Routing &mdash; #20900 🔴
  - [x] Configuration &mdash; #20908 🔴
  - [x] Dependency injection &mdash; #20920 🔴
  - [x] Startup &mdash; #22279 🔴
  - [x] Environments &mdash; #20929 🔴
  - [x] Logging &mdash; #20929 🔴
  - [x] Handle errors &mdash; #21601 🔴
  - [x] SignalR &mdash; #21620 🔴
  - [x] Static files &mdash; #21331 🔴
- [x] Components
  - [x] Overview &mdash; #22053 🔴
  - [x] Cascading values and parameters &mdash; #21366 🔴
  - [x] Data binding &mdash; #21678 🔴
  - [x] Event handling &mdash; #21755 🔴
  - [x] Lifecycle &mdash; #21822 🔴
  - [x] Rendering &mdash; #21766 🔴
  - [x] Component virtualization &mdash; #21623 🔴
  - [x] Templated components &mdash; #21666 🔴
  - [x] CSS isolation 🔴
  - [x] Prerender and integrate components &mdash; #21881 🔴
  - [x] Component libraries &mdash; #21946
- [x] Globalization and localization &mdash; #22525 🔴
- [x] Layouts &mdash; #21636 🔴
- [x] Forms and validation &mdash; #22106 🔴
- [x] File uploads &mdash; #22161 🔴
- [x] JS interop
  - [x] Overview
  - [x] Call JavaScript from .NET &mdash; #22229 🔴
  - [x] Call .NET from JavaScript &mdash; #22304 🔴
- [x] Call a web API from WebAssembly &mdash; #22410 🔴
- [ ] Security and Identity &mdash; See :point_right: #20708, #19807, and #19226. I placed an ellipsis in the `App` component example INCLUDE(s) to cover 6.0 churn. See: https://github.com/dotnet/AspNetCore.Docs/issues/23696
  - [ ] Overview &mdash; Either here or in the Blazor Server *Overview*: #22405
  - [ ] Blazor WebAssembly &mdash; See :point_right: #20512 & https://github.com/dotnet/AspNetCore.Docs/issues/21816#issuecomment-803152089. Also, consider surfacing a smidgen of WebAssembly "sandbox" content in a new section with a cross-link or two (e.g., https://webassembly.github.io/spec/core/intro/introduction.html#security-considerations, https://webassembly.org/docs/security/). It would be nice to clarify that WebAssembly (as a non-MS technology) is constantly undergoing security patching, and there is a place (I'll find the link) where devs can see potential security vulnerabilities, work on WebAssembly security, and patches being put into place.
    - [ ] Overview &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21741 (cross-link to an Azure doc from all of our AAD/B2C docs)
    - [ ] Standalone with Authentication library &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21510, https://github.com/dotnet/aspnetcore/issues/30068, https://github.com/dotnet/AspNetCore.Docs/issues/23712
    - [ ] Standalone with Microsoft Accounts
    - [ ] Standalone with AAD
    - [ ] Standalone with AAD B2C &mdash; "custom user flows" language https://github.com/dotnet/AspNetCore.Docs/issues/21144. Take a look at signup-signin Azure UI per https://github.com/dotnet/aspnetcore/issues/39401#issuecomment-1009511236.
    - [ ] Hosted with AAD &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21983
    - [ ] Hosted with AAD B2C &mdash; "custom user flows" language https://github.com/dotnet/AspNetCore.Docs/issues/21144; see :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21640. Take a look at signup-signin Azure UI per https://github.com/dotnet/aspnetcore/issues/39401#issuecomment-1009511236.
    - [ ] Hosted with Identity Server &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21842 and also see https://github.com/dotnet/AspNetCore.Docs/issues/22753 (Was it just VS not creating the project with the correct Identity config to run OOB when deployed to Azure directly from VS?) Consider splitting out custom domains from IdS, and check on the B1 requirement IdS-only scnearios (it might not be a requirement in that case). See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/24067#issuecomment-980033383.
    - [ ] Additional scenarios - Look for split-out opportunities. Possibly add an introductory ToC with section descriptions. Consider if the *Unauthenticated or unauthorized web API requests in an app with a secure default client* section could be better cross-linked/mentioned explicitly in the individual WASM security topics (https://github.com/dotnet/AspNetCore.Docs/issues/22217 and https://github.com/dotnet/AspNetCore.Docs/pull/22220). Consider making an INCLUDE out of the prerendering guidance and shooting a copy of that into the *Prerendering and Integration* topic (https://github.com/dotnet/AspNetCore.Docs/issues/23441).
    - [ ] AAD groups and roles &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/20856; do we need a Blazor Server pivot, or can we cross-link? See :point_right: #22422
    - [ ] Graph API &mdash; See :point_right: #22657
  - [ ] Blazor Server
    - [ ] Overview &mdash; Review, work was done on :point_right: https://github.com/dotnet/AspNetCore.Docs/pull/20119 and mention https://github.com/dotnet/AspNetCore.Docs/issues/19797#issuecomment-779294104.
    - [ ] Threat mitigation
    - [ ] Additional scenarios &mdash; Review, work was done on :point_right: https://github.com/dotnet/AspNetCore.Docs/pull/20119. Also, see :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/22122.
  - [ ] Content Security Policy
- [x] State management 🔴
- [ ] Debug WebAssembly &mdash: Resolve :point_right: #23373 ... and we'll probably need to show full files everywhere we refer to `tasks.json`/`launch.json` due to https://github.com/OmniSharp/omnisharp-vscode/issues/4542. See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/23777
- [x] Lazy load assemblies with WebAssembly &mdash; #22603 🔴
- [x] WebAssembly performance &mdash; #23135
- [x] Test components 🔴
- [ ] Progressive Web Applications &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21100
- [ ] Host and deploy: Apache for hosted WASM/sub-app scenarios might need more work. See 👉 https://github.com/dotnet/AspNetCore.Docs/pull/24519#issuecomment-1008480134.
  - [ ] Overview &mdash; Perhaps for *Routing*, too, but clarify the navigation behavior of `NavigationManager.NavigateTo`. See :point-right: https://github.com/dotnet/AspNetCore.Docs/pull/22146#issuecomment-827802596.
  - [ ] Blazor WebAssembly &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/19979 and https://github.com/dotnet/AspNetCore.Docs/issues/21829#issuecomment-803192673
  - [ ] Blazor Server &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/21226
  - [x] Configure the Trimmer &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/21480 🔴
- [ ] Blazor Server and EF Core
- [x] Advanced scenarios &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/21768 🔴
- [ ] Scaffolding topic sections that pertain to Blazor &mdash; See :point_right: https://github.com/dotnet/AspNetCore.Docs/issues/8434#issuecomment-808893911
- [ ] File downloads topic: Use fully-working examples, place code into snippet sample apps, and see the cross-linked discussion and bits mentioned in https://github.com/dotnet/AspNetCore.Docs/pull/24560#issuecomment-1011301931.
- [ ] Images topic: Use fully-working examples.

**Comment by guardrex** [comment]
Closing out and moving to a new streamlined issue for '22 ...

https://github.com/dotnet/AspNetCore.Docs/issues/24615
